File: [20190106-20190109]/java/revision2-form/src/main/java/com/edugroupe/revision2form/metier/Vin.java
package com.edugroupe.revision2form.metier;

import java.util.Set;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.ManyToMany;
import javax.persistence.ManyToOne;

import com.fasterxml.jackson.annotation.JsonIgnore;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter @Setter @NoArgsConstructor @AllArgsConstructor
@Entity
public class Vin {
	@Id @GeneratedValue(strategy = GenerationType.IDENTITY)		private int id;
																private String nom;
																private int annee;
	@ManyToOne 													private Terroir terroir;
	@ManyToMany @JsonIgnore										private Set<Caracteristique> caracteristiques;
	
}

File: [20190106-20190109]/angular/cavegemini/src/app/metier/vin.ts
import { Terroir } from './terroir';
import { Caracteristique } from './caracteristique';

export class Vin {
    /*public id : number;

    constructor(id : number) {
        this.id = id;
    }*/
    constructor (   public id: number,
                    public nom: string,
                    public annee: number,
                    public terroir?: Terroir,
                    public caracteristiques?: Caracteristique[]) {}


    
    
}